apps/server/src/modules/executor/executor.service.ts

```diff
@@ -1,17 +1,106 @@
 import { Injectable } from '@nestjs/common';
-import { Observable } from 'rxjs';
 import { EXECUTOR_URLS } from '~constants';
 import { HttpClientService } from '~utils/http-client';
-import { LanguageDto } from './dto/language.dto';
+import {
+  LanguageDto,
+  AvailableLanguageDto,
+  PossibleLanguageDto,
+  ExecuteCodeDto,
+  ExecuteCodeResultDto,
+  PossibleLanguagesResponseDto,
+  AvailableLanguagesResponseDto,
+} from './dto';
+import { isAxiosError } from 'axios';
+import { EResponseStatus, IBaseErrorResponse, TBaseResponse } from '~types';
 
 @Injectable()
 export class ExecutorService {
   constructor(private readonly httpClient: HttpClientService) {}
 
+  async executeCode(
+    dto: ExecuteCodeDto,
+  ): Promise<TBaseResponse<ExecuteCodeResultDto>> {
+    const url = `http://${process.env.CODE_EXECUTOR_HOST}:${process.env.CODE_EXECUTOR_PORT}${EXECUTOR_URLS.execute}`;
+
+    try {
+      const response = await this.httpClient.post<
+        ExecuteCodeResultDto,
+        ExecuteCodeDto
+      >(url, dto);
+
+      return {
+        status: EResponseStatus.Success,
+        data: response.data,
+      };
+    } catch (error) {
+      return this.handleError(error);
+    }
+  }
+
   // TODO добавить валидация данных в ответе
-  getAllLanguages(): Observable<LanguageDto[]> {
-    const url = `http://${process.env.CODE_EXECUTOR_HOST}:${process.env.CODE_EXECUTOR_PORT}${EXECUTOR_URLS.allLanguages}`;
+  async getPossibleLanguages(): Promise<
+    TBaseResponse<PossibleLanguagesResponseDto>
+  > {
+    const url = `http://${process.env.CODE_EXECUTOR_HOST}:${process.env.CODE_EXECUTOR_PORT}${EXECUTOR_URLS.packages}`;

```

Comment:
Может базовый путь сохранить на уровне параметра класса, чтобы не собирать его в каждом методе?

```suggestion
    const url = `${this._baseUrl}${EXECUTOR_URLS.packages}`;

```

Reply:
Отличная идея. Сделал